lib/pdn/pdn.c

```diff
@@ -692,6 +709,21 @@ static void on_cfun(int mode, void *ctx)
 	}
 
 	if (mode == MODEM_CFUN_POWER_OFF) {
+		k_mutex_lock(&list_mutex, K_FOREVER);
+		SYS_SLIST_FOR_EACH_CONTAINER_SAFE(&pdn_contexts, pdn, tmp, node) {
+			if (pdn->context_id == 0) {
+				/* Default context is not destroyed */
```

Comment:
When modem is switched to minimum functionality mode (`+CFUN=0`), also the configuration for the default context is lost. So, I'm not sure this is the right way to handle default context in the library.

Example:

```
mosh:~$ at at+cgdcont=0,"IP","foobar"
OK

mosh:~$ at at+cgdcont?
+CGDCONT: 0,"IP","foobar","",0,0
OK

mosh:~$ at at+cfun=0
OK

mosh:~$ at at+cgdcont?
OK
```

Reply:
I think it is not intended for the user to change the default APN themselves, we have `CONFIG_PDN_DEFAULTS_OVERRIDE` for that. When using this config the configuration is automatically re-applied on CFUN=0. So it would make sense to still have the callback.

When would you suggest the user to set the callback again if this exception is not made? We cannot do it (without extra changes) in the `PDN_EVENT_CTX_DESTROYED` event because we are in the loop deleting contexts. We cannot use our own CFUN=0 hook as there is no way to tell in which order they will be handled.

Reply:
I'm not that familiar with the PDN library and I did not remember that the default context is handled differently. I think you're right and the current behavior is correct.